functions/safe_filename.pp

```diff
@@ -0,0 +1,70 @@
+# @summary Returns a string that is safe for firewalld filenames
+#
+# @example Regular Filename
+#   $filename = 'B@d Characters!'
+#   firewalld::safe_filename($orig_string)
+#
+#   Result => 'B_d_Characters_'
+#
+# @example Filename with Options
+#   $filename = 'B@d Characters!.txt'
+#   firewalld::safe_filename(
+#     $filename,
+#     {
+#       'replacement_string' => '@@',
+#       'file_extension'     => '.txt'
```

Comment:
In this example, is replacing the bad character `@` with `@@` a desirable result?

Reply:
@op-ct Hmm....you're probably right, that should be an error if you try to do something that will create an invalid filename. Will patch.